devtools/src/main/scala/kalix/devtools/impl/KalixProxyContainer.scala

```diff
@@ -95,7 +91,10 @@ class KalixProxyContainer private (image: DockerImageName, config: KalixProxyCon
     containerLogger.info("Starting Kalix Proxy Server container...")
     containerLogger.info("Using proxy image [{}]", image)
     containerLogger.info("KalixProxyContainer config : {}", config)
+
     Testcontainers.exposeHostPorts(userFunctionPort)
+    config.pubsubEmulatorPort.foreach(Testcontainers.exposeHostPorts(_))
```

Comment:
Is this actually needed? I thought this was only needed to expose ports from the testcontainters to outside? but since the pubsub in this case is expect to run externally to the testcontainers.. is this needed? or maybe I misunderstood something.

Reply:
`exposeHostPorts` is used to expose host ports to the process running inside testcontainer. 

From the perspective of the kalix proxy, both pubsub and user function are running on the host. Although pubsub runs in a container, it's accessible from the host.

I can check if that's really necessary, but this is how the KalixTestKit configures the access to the user function as well.

Reply:
yeah, you're right. I got it wrong previously. Thanks :)